cloud/store/service.go

```diff
@@ -38,16 +40,28 @@ type client interface {
 }
 
 func NewService(store *Store, client client) *Service {
-	return &Service{store: store, cl: client}
+	return &Service{store: store, cl: client, log: store.log}
 }
 
 // / RAFT-TODO Documentation
 func (s *Service) Open(ctx context.Context, db Indexer) error {
+	s.log.Info("bootstrapping started")
 	s.store.SetDB(db)
 	return s.store.Open(ctx)
 }
 
 func (s *Service) Close(ctx context.Context) (err error) {
+	s.log.Info("shutting down raft sub-system ...")
+
+	// non-voter can be safely removed, as they don't partake in RAFT elections
+	if !s.store.IsVoter() {
+		s.log.Info("removing this node from cluster prior to shutdown ...")
+		if err := s.Remove(ctx, s.store.ID()); err != nil {
+			s.log.Error("remove this node from cluster: " + err.Error())
+		} else {
+			s.log.Info("successfully removed this node from the cluster.")
+		}
+	}
```

Comment:
**thought (if-minor):** is it needed ? given that it will close anyway 🤔

Reply:
This opt saves network traffic by eliminating the need for the the leader to contact offline non-voters